projects/RabbitMQ.Client/client/impl/SocketFrameHandler.cs

```diff
@@ -122,11 +134,12 @@ class SocketFrameHandler : IFrameHandler
                     throw;
                 }
             }
+
             _reader = new BufferedStream(netstream, _socket.Client.ReceiveBufferSize);
             _writer = new BufferedStream(netstream, _socket.Client.SendBufferSize);
 
             WriteTimeout = writeTimeout;
-            _frameWriter = Task.Run(WriteFrameImpl);
+            _writerTask = Task.Run(WriteLoop, CancellationToken.None);
```

Comment:
I can't remember but I was thinking there was once a `Delegate allocation from method group` problem hiding here or is that gone in recent compiler optimizations?

Reply:
I love that you explicitly pass `CancellationToken.None`

Reply:
As far as I know, yes it does allocate, that's correct. The thing though is any instance method wrapped in a delegate does. 
Hence changing it to not use a  method group doesn't eliminate the allocation, it just does allocate a different one. Also caching upfront doesn't help, as we'd just allocate then there instead of here. (Only benefits if it would be called multiple times, but normally it won't)

Reply:
Yeah true it is not a hot path because it only gets kicked off once